backend/server/routers/collection.py

```diff
@@ -140,6 +140,12 @@ async def unassociate_data_source_from_collection(
             collection_name=request.collection_name,
             data_source_fqn=request.data_source_fqn,
         )
+        VECTOR_STORE_CLIENT.delete_data_point_vectors(collection_name=request.collection_name,
+                data_point_vectors=VECTOR_STORE_CLIENT.list_data_point_vectors(
+                    collection_name=request.collection_name,
+                    data_source_fqn=request.data_source_fqn
+            )
+        )
```

Comment:
This change itself is fine I guess

Ideally, 
- delete should be implemented directly instead of having to fetch the data points first
- Our vector store implementations have a hidden limit of 1e6 on points scroll. We should not have any limits, rather pagination with a batch size
- This deletion should also be async worker job because the operation might take much more longer

Reply:
That being said this change might still be useful for collections with less than 10k data points

Reply:
> * This deletion should also be async worker job because the operation might take much more longer

can use the same process_pool here used in https://github.com/truefoundry/cognita/pull/321 , with the same caveats.
If we already have 2 use cases for this, better to think about queue & worker based approach